algoliasearch-apache/src/main/java/com/algolia/search/ApacheHttpRequester.java

```diff
@@ -31,14 +32,13 @@
  * implementation of {@link HttpRequester} It takes an {@link HttpRequest} as input. It returns an
  * {@link HttpResponse}.
  */
-final class ApacheHttpRequester implements HttpRequester {
+public final class ApacheHttpRequester implements HttpRequester {
 
   private final CloseableHttpAsyncClient asyncHttpClient;
   private final RequestConfig requestConfig;
   private final ConfigBase config;
 
-  ApacheHttpRequester(@Nonnull ConfigBase config) {
-
+  public ApacheHttpRequester(@Nonnull ConfigBase config) {
```

Comment:
nit: We could call `ApacheHttpRequester(@Nonnull ConfigBase config, @Nonnull HttpAsyncClientBuilder builder)` with the builder created in place in `ApacheHttpRequester(@Nonnull ConfigBase config)` to avoid duplicated code.

Reply:
I wanted to do it at first but then abandoned because I needed to pass the big ternary in a `this( ... );` statement which must be the first immediate statement in the constructor to be able to chain them. But after retrying it, it seems kinda clean anyway so I've did it in the end.